### PR TITLE
fix: put processes in class for state

### PR DESCRIPTION
**Pull request by bodinsamuel**

### CODE REVIEW

**Comment by Haroenv** [comment]
Big change :D what's the pros/cons?

**Comment by bodinsamuel** [comment]
Having more and more "floating" state was bothering me + passing all params for each function starts to be too verbose.
Mostly a coding style reason, I'm fine with functional with it's very simple and portable function, here they are all tied together.

**Comment by bodinsamuel** [comment]
Note: it's easier to review with `Hide whitespace` checked

File: src/watch.ts [hunk 1]
```diff
@@ -12,189 +13,196 @@ import { datadog } from './utils/datadog';
...
+   *
+   * @param stateManager - The state manager.
+   * @param mainIndex - Algolia index manager.
+   */
+  async watch(): Promise<boolean | undefined> {
+    const { seq } = await this.stateManager.get();
+    const changesConsumer = this.changesConsumer;
+    if (!changesConsumer) {
+      log.warn('Can not watch without a consumer');
       return;
```
Comment (Haroenv) [inline src/watch.ts:85]: if we return `undefined`, is the eslint rule disabling still needed?

```suggestion
      return undefined;
```

**Review by Haroenv — APPROVED** [review]
code itself lgtm! would like to see if there's a way where the eslint rule doesn't need to be disabled

File: src/watch.ts (hunk 1 above)
Comment (bodinsamuel) [reply src/watch.ts:85]: There is an issue for that. we could disable the rule entirely for TS though 
https://github.com/typescript-eslint/typescript-eslint/issues/1277

**Pull request merged**